File: rust/src/models/customer_portal_token_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CustomerPortalTokenResponse {
    /// Base URL of the customer portal
    pub portal_url: String,

    /// JWT token for portal access
    pub token: String,
}

impl CustomerPortalTokenResponse {
    pub fn new(portal_url: String, token: String) -> Self {
        Self { portal_url, token }
    }
}

File: typescript/src/models/customerPortalTokenResponse.ts
// this file is @generated

export interface CustomerPortalTokenResponse {
  /** Base URL of the customer portal */
  portalUrl: string;

  /** JWT token for portal access */
  token: string;
}

export const CustomerPortalTokenResponseSerializer = {
  _fromJsonObject(object: any): CustomerPortalTokenResponse {
    return {
      portalUrl: object["portal_url"],
      token: object["token"],
    };
  },

  _toJsonObject(self: CustomerPortalTokenResponse): any {
    return {
      portal_url: self.portalUrl,
      token: self.token,
    };
  },
};
